OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -244,7 +249,7 @@ public override bool HandleKeyPress(KeyInput e)
 						Game.Debug("Selected across screen");
 					else
 					{
-						newSelection = World.ActorMap.ActorsInWorld().Where(cond);
+						newSelection = World.ActorMap.ActorsInWorld().Where(x => x.HasTrait<Selectable>() && x.Trait<Selectable>().Info.Selectable && cond(x));
 						Game.Debug("Selected across map");
 					}
```

Comment:
I was thinking about putting the select-by-type logic inside a separate function since these seem similar enough to allow for exporting the functionality